Source/Core/Core/IOS/ES/Formats.cpp

```diff
@@ -600,32 +605,34 @@ bool SharedContentMap::WriteEntries() const
          HLE::FS::ResultCode::Success;
 }
 
-static std::pair<u32, u64> ReadUidSysEntry(const HLE::FS::FileHandle& file)
+static std::pair<u32, u64> ReadUidSysEntry(HLE::FSDevice& fs, u64 fd, u64* ticks)
 {
   u64 title_id = 0;
-  if (!file.Read(&title_id, 1))
+  if (fs.Read(fd, &title_id, 1, ticks) != sizeof(title_id))
```

Comment:
Can't a constant be used to indicate an invalid read or an invalid `title_id` rather than relying on `sizeof`?

Reply:
sizeof(title_id) *is* that constant. I could hardcode 4 but I think that's worse than using sizeof.

Reply:
I see, I just find odd that the `Read` function can have an error state and it's not being used to handle the error. I understand the reasoning behind this use of `sizeof` in the case of partially read. Maybe that's just me but I find it strange to read.

Reply:
If there is an error, Read() returns a negative value so the comparison will be false in that case. Or am I misunderstanding your comment?

Reply:
I was just saying that having `if (Read(..., &var, ...) != READ_SIZE)` rather than `if (Read(..., &var, ...) < 0)`/`if (Read(..., &var, ...) == ERROR` seems a bit odd to read to me.

Feel free to disregard that comment.

Reply:
Oh, I see what you mean. But if we do `if (Read(..., &var, ...) < 0)` we still also need to check for partial reads, so we'd have to do something like

```cpp
const int ret = Read(...);
if (ret < 0 || ret != READ_SIZE)
  ...
```
which seems redundant to me, since READ_SIZE is positive.